### PR TITLE
add environment.goals and environmentalGoal.csv

### LINKED ISSUE #415: Climate change objectives
# Background

This issue relates to the following CoST IDS elements proposed in the CoST IDS/OC4IDS review:

<details>

<summary> Climate objective (identification) </summary>

## Climate objective (identification)

**Module:** Climate finance
**Indicator:** Project theme

### Disclosure format

> Disclose the main climate objective that the project addresses.   
>
> Select from a list:  
>
> Eg:  
>
> * mitigation
> * adaptation
> * cross-cutting

### OC4IDS mapping

> **Project level:** Add the relevant code from the climateChangeGoal codelist to the `environment.climateChangeGoals` array

</details>

I suggested incorporating a project's climate change objectives into a broader modelling of sustainability objectives based on the EU's Sustainable Finance Taxonomy, however CoST's consultant that proposed the climate finance data points advised against aligning with the EU taxonomy due to sensitives in other parts of the world.

# Proposal

Add a field and codelist to disclose whether a project's climate change goals are mitigation, adaptation or both.

Add the following field to the `environment` object proposed in #393:

Path | Title | Description | Type (format) | Codelist
-- | -- | -- | -- | --
`environment.goals` | Environmental goals | The environmental goals pursued, from the closed environmentalGoal codelist. | Array | environmentalGoal

Add a environmentalGoal codelist with the following codes:

Codelist | Code | Title | Description
-- | -- | -- | --
environmentalGoal | climateChangeMitigation | Climate change mitigation | Reduction of emissions or enhancement of the sinks of greenhouse gases.
environmentalGoal | climateChangeAdaptation | Climate change adaptation | Adjustment to actual or expected climate and its effects, in order to moderate harm or exploit beneficial opportunities.

# Example

```json
{
  "environment": {
    "goals": ["mitigation"]
  }
}
```

# Sources

* [OCDS Sustainability Extension](https://extensions.open-contracting.org/en/extensions/sustainability/master/) - field names, titles and descripitons
* [IPCC Glossary](https://apps.ipcc.ch/glossary/) - code descriptions

cc @evelyndinora, @mgraca-prado

### PR DESCRIPTION
**Pull request by odscjen**
**Related issues**

closes #415 

**Description**

**Merge checklist**

- [x] [Log your changes](https://ocds-standard-development-handbook.readthedocs.io/en/latest/standard/contributing.html#logging-changes)

If there are changes to `project-schema.json` or `project-package-schema.json`:

- [x] Update the examples:
  - [ ] `docs/examples/example.json`
  - [ ] `docs/examples/blank.json`
- [x] Run `./manage.py pre-commit` to update `docs/_static/i8n.csv`

If you added a new definition to the schema, run `./manage.py pre-commit`.

If you added a new codelist:

  - [x] Add an entry to `docs/reference/codelists.md`

### CODE REVIEW

**Comment by odscjen** [comment]
CI / build (push) appears to be stuck, Details is showing

> Requested labels: ubuntu-latest
Job defined at: open-contracting/infrastructure/.github/workflows/ci.yml@refs/heads/415_climate_objectives
Waiting for a runner to pick up this job...
Job is waiting for a hosted runner to come online.

and has been running for almost 18 minutes now so I'm leaving it for the night and hopefully the runner will come online at some point!

@duncandewhurst if it doesn't I'll try and sort it when I'm back in on Thursday. If it does and there's still errors just leave it and I'll sort them on Thursday. If it does and the build passes then please review :)

File: mapping/sustainability.yaml [hunk 1]
```diff
@@ -486,12 +486,12 @@
         - mitigation
         - adaptation
         - cross-cutting
-    mapping: 'Project level: Add the relevant code from the climateChangeGoal codelist to the `environment.climateChangeGoal` array'
+    mapping: 'Project level: Add the relevant code from the environmentalGoal codelist to the `environment.goals` array'
```
Comment (duncandewhurst) [inline mapping/sustainability.yaml]:
```suggestion
    mapping: 'Project level: Add the relevant codes from the environmentalGoal codelist to the `environment.goals` array'
```

File: schema/project-level/project-schema.json [hunk 2]
```diff
@@ -296,6 +296,31 @@
...
       "minLength": 1
+    },
+    "environment": {
+      "title": "Environment",
+      "description": "Information about the environmental and climate impact of the project and actions the project will take to mitigate potentially negative impacts on adjacent populations and habitats.",
+      "type": "object",
+      "properties": {
+        "goals": {
+          "title": "Environmental goals",
+          "description": "The environmental goals pursued, using the closed environmentalGoal codelist.",
```
Comment (duncandewhurst) [inline schema/project-level/project-schema.json]: Add link to codelist reference.

```suggestion
          "description": "The environmental goals pursued, using the closed [environmentalGoal codelist](https://standard.open-contracting.org/infrastructure/{{version}}/{{lang}}/reference/codelists/#environmentalgoal).",
```

**Review by duncandewhurst — CHANGES_REQUESTED** [review]
Happy for you to request a review from James once you've resolved my comments.

File: schema/project-level/project-schema.json [hunk 3]
```diff
@@ -296,6 +296,31 @@
       ],
       "default": "en",
       "minLength": 1
+    },
+    "environment": {
+      "title": "Environment",
+      "description": "Information about the environmental and climate impact of the project and actions the project will take to mitigate potentially negative impacts on adjacent populations and habitats.",
```
Comment (jpmckinney) [inline schema/project-level/project-schema.json]: "Information about ... **the**"

```suggestion
      "description": "Information about the environmental and climate impacts of the project and the actions the project will take to mitigate potentially negative impacts on adjacent populations and habitats.",
```

File: schema/project-level/project-schema.json [hunk 4]
```diff
@@ -296,6 +296,31 @@
...
+      "type": "object",
+      "properties": {
+        "goals": {
+          "title": "Environmental goals",
+          "description": "The environmental goals pursued, using the closed [environmentalGoal codelist](https://standard.open-contracting.org/infrastructure/{{version}}/{{lang}}/reference/codelists/#environmentalgoal).",
```
Comment (jpmckinney) [inline schema/project-level/project-schema.json]: We usually put codelist outside the link, though there are some exceptions (which can be corrected).

```suggestion
          "description": "The environmental goals pursued, using the closed [environmentalGoal](https://standard.open-contracting.org/infrastructure/{{version}}/{{lang}}/reference/codelists/#environmentalgoal) codelist.",
```

**Review by jpmckinney — COMMENTED** [review]
Two small edits, then good to merge.

**Pull request merged**